vm-virtio/src/block.rs

```diff
@@ -767,6 +767,7 @@ impl<T: DiskFile> BlockEpollHandler<T> {
             }
         }
 
+        epoll::close(epoll_fd).map_err(DeviceError::EpollClose)?;
```

Comment:
Problem with this approach is that you're gonna close the file descriptor only if things are Ok(()). You want to make sure that even if this function returns earlier than expected (because of an Err()), the file descriptor will be closed.

One simple way to do it would be to get a `File` from the `epoll_fd` `RawFd`. The created `File` would be dropped at the end of the function and `close()` will be automatically called on this `File`.

There's also another way, and we should consider this for the long term plan. There's a new crate on `rust-vmm` called `event_manager`, which abstract a bunch of the complexity and duplication related to epoll events. We want to switch to this eventually, and I think this crate won't leak any file descriptor based on the way it's implemented.
https://github.com/rust-vmm/event-manager

Reply:
/cc @rbradford @sameo

Reply:
@likebreath please go ahead with the quick solution of using `let epoll_file = unsafe{ File::from_raw_fd(epoll_fd) };`. Every place where the code was expecting `epoll_fd`, you can use `epoll_file.as_raw_fd()`. This way we make sure at the end of the scope, the `File` will take care of closing the file descriptor.
I opened an issue https://github.com/cloud-hypervisor/cloud-hypervisor/issues/1183 to track the move to `event-manager` crate from rust-vmm.